stl/inc/flat_map

```diff
@@ -0,0 +1,335 @@
+#pragma once

```

Comment:
The usual header stuff like `_STL_COMPILER_PREPROCESSOR` and guard against `new` keyword redefinition.
Some of this is conventional: even if this header does not use `new`, this should be present

Reply:
Added.

Reply:
Hmm, I find <flat_set> header also lack these `new` guards
https://github.com/microsoft/STL/blob/1f4920713ecfed75de12eeca8375e4cab4e1ba56/stl/inc/flat_set

Reply:
Should be added there too then